### PR TITLE
Feat/simple backups support

### PR DESCRIPTION
**Pull request by ka-myl**
Adds support for simple backup option for servers.

### CODE REVIEW

File: internal/server/server.go [hunk 1]
```diff
@@ -304,3 +320,36 @@ func UpdateServerTags(service *service.Service, serverUUID string, oldTags, newT
...
+func ValidateServerAndStorageBackupRules(service *service.Service, d *schema.ResourceData) error {
+	if _, ok := d.GetOk("simple_backup"); ok {
+		if attr, ok := d.GetOk("storage_devices"); ok {
+			sds := attr.(*schema.Set).List()
+
+			for _, i := range sds {
+				e := i.(map[string]interface{})
+				id := e["storage"]
+
+				sd, err := service.GetStorageDetails(&request.GetStorageDetailsRequest{
```
Comment (ka-myl) [inline internal/server/server.go:334]: Doing a request for each storage is imo a bit heavy, but I haven't found a better way to validate this. If you have some other ideas please let me know

File: internal/server/server.go [hunk 2]
```diff
@@ -304,3 +320,36 @@ func UpdateServerTags(service *service.Service, serverUUID string, oldTags, newT
 
 	return AddNewServerTags(service, serverUUID, newTags)
 }
+
+// ValidateServerAndStorageBackupRules checks if the server and (to be) attached storages have conflicting backup rules.
+func ValidateServerAndStorageBackupRules(service *service.Service, d *schema.ResourceData) error {
+	if _, ok := d.GetOk("simple_backup"); ok {
```
Comment (Darep) [inline internal/server/server.go:326]: personally would reverse the check here so we immediately return nil in this case (so we don't have so many blocks/so much indentation)

Comment (ka-myl) [reply internal/server/server.go]: Removed this whole function. It could lead to errors on subsequent updates with some configs.

**Comment by kaminek** [comment]
@ka-myl can you rebase please and fix the conflicts?

File: upcloud/resource_upcloud_server.go [hunk 3]
```diff
@@ -305,6 +307,29 @@ func resourceUpCloudServer() *schema.Resource {
 					},
 				},
 			},
+			"simple_backup": {
+				Description:   "Simple backup schedule configuration",
+				Type:          schema.TypeList,
```
Comment (kaminek) [inline upcloud/resource_upcloud_server.go:312]: please use `TypeSet` that represents unordered collection

File: upcloud/resource_upcloud_server.go [hunk 4]
```diff
@@ -385,6 +411,17 @@ func resourceUpCloudServerRead(ctx context.Context, d *schema.ResourceData, meta
...
 		_ = d.Set("firewall", false)
 	}
+	if server.SimpleBackup != "no" {
+		p := strings.Split(server.SimpleBackup, ",")
+		simpleBackup := map[string]interface{}{
+			"time": p[0],
+			"plan": p[1],
+		}
+
+		_ = d.Set("simple_backup", []map[string]interface{}{simpleBackup})
```
Comment (kaminek) [inline upcloud/resource_upcloud_server.go:421]: just `simpleBackup`

File: upcloud/resource_upcloud_server.go [hunk 5]
```diff
@@ -385,6 +411,17 @@ func resourceUpCloudServerRead(ctx context.Context, d *schema.ResourceData, meta
...
+	} else {
```
Comment (kaminek) [inline upcloud/resource_upcloud_server.go:422]: not needed, default is `nil`

File: upcloud/resource_upcloud_server.go [hunk 6]
```diff
@@ -390,6 +416,15 @@ func resourceUpCloudServerRead(ctx context.Context, d *schema.ResourceData, meta
 	} else {
 		_ = d.Set("firewall", false)
 	}
+	if server.SimpleBackup != "no" {
```
[the remark this answers is not in the document]
Comment (moitias) [reply upcloud/resource_upcloud_server.go:419]: Yep, that's more of a -libs thing I feel!

**Pull request merged**